pynicotine/plugins/core_commands/__init__.py

```diff
@@ -26,39 +26,637 @@ def __init__(self, *args, **kwargs):
+            "say": {
+                "callback": self.say_chat_command,
+                "description": _("Say message in specified chat room"),
+                "disable": ["cli"],
+                "group": _("Chat Rooms"),
+                "usage": ["<room>", "<message..>"]
+            },
             "close": {
-                "description": "Close private chat",
                 "aliases": ["c"],
-                "disable": ["cli"],
-                "group": "Private Chat",
                 "callback": self.close_command,
-                "usage_chatroom": ["<user>"],
+                "description": _("Close private chat"),
+                "disable": ["cli"],
+                "group": _("Private Chat"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "leave": {
+                "aliases": ["l"],
+                "callback": self.leave_command,
+                "description": _("Leave chat room"),
+                "disable": ["cli"],
+                "group": _("Chat Rooms"),
+                "usage": ["<room>"],
+                "usage_chatroom": ["[room]"]
+            },
+            "now": {
+                "callback": self.now_playing_command,
+                "description": _("Display the Now Playing script's output"),
+                # "disable": ["cli"],
+                "group": _("Now Playing"),
+            },
+            "add": {
+                "aliases": ["buddy"],
+                "callback": self.add_buddy_command,
+                "description": _("Add user to buddy list"),
+                "group": _("Users"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "rem": {
+                "aliases": ["unbuddy"],
+                "callback": self.remove_buddy_command,
+                "description": _("Remove user from buddy list"),
+                "group": _("Users"),
+                "usage": ["<buddy>"],
+                "usage_private_chat": ["[buddy]"]
+            },
+            "ban": {
+                "callback": self.ban_user_command,
+                "description": _("Stop connections from user"),
+                "group": _("Users"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "unban": {
+                "callback": self.unban_user_command,
+                "description": _("Remove user from ban list"),
+                "group": _("Users"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "block": {  # new
+                "callback": self.block_user_ip_command,
+                "description": _("Stop all connections from same IP as user"),
+                "group": _("Network Filters"),
+                "usage": ["<user>"],
+                "usage_private_chat": ["[user]"]
+            },
+            "unblock": {  # new
```

Comment:
Suggest unban-ip

Reply:
This command does not (currently) support accepting an IP address as argument, so the proposed name would be misleading.

Reply:
I've moved away from the "block" terminology in the GUI over the past year or so.

Reply:
Agreed, it's not reasonable to expect the user to deal with a whole range of commands each with subtle differences.

It is going to be necessary to allow for more flexible arguments, even if this involves making some assumptions, such as:

    If we specify an ip_address, then the intention is to ban/unban all usernames associated with that address.
    If we specify a username, then the intention may be inferred also to ban/unban all users at that same address.

The <user> argument now validates on either a "username" or an "ip_address" type of input for both the `/ban` and `/unban `commands, but a core functionality is missing from networkfilter.py to enable ban/unban without knowing a username (yet it should be possible because `ipblocklist` does accept a zero-length string for the username, and perhaps we can try look up an associated username for `banlist` if we have an active/current connection).

Reply:
Resolved, both username-based and IP-based bans now get removed with a single `/unban` command.